src/query/storages/cache/src/providers/memory_bytes_cache.rs

```diff
@@ -0,0 +1,86 @@
+// Copyright 2022 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use common_cache::BytesMeter;
+use common_cache::Cache;
+use common_cache::DefaultHashBuilder;
+use common_cache::LruCache;
+use common_exception::Result;
+use opendal::Object;
+use parking_lot::RwLock;
+
+use crate::CacheSettings;
+use crate::ObjectCacheProvider;
+
+type BytesCache = RwLock<LruCache<String, Arc<Vec<u8>>, DefaultHashBuilder, BytesMeter>>;
+
+/// Memory LRU cache for bytes(Vec<u8>).
+pub struct MemoryBytesCache {
+    lru: BytesCache,
```

Comment:
Although it is currently Lru, I think we could consider using a name that doesn't point so much to a specific algorithm in MemoryBytesCache & MemoryItemsCache.

Reply:
Memory cache should always LRU.